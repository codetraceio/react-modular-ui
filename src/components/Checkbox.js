import React from 'react';

import settings from '../settings';

import AbstractComponent from './AbstractComponent';
import Icon from './Icon';

export default class Checkbox extends AbstractComponent {
  getModifierObject() {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      disabled: this.props.disabled
    };
  }

  isChecked() {
    return this.props.checked || false;
  }

  onChange(e) {
    if (typeof this.props.onChange === 'function') {
      this.props.onChange(!this.props.checked, e);
    }
  }

  render() {
    if (settings.isBackend()) {
      return (
        <div
          className={this.getBlockName('checkbox', this.getModifierObject())}
          role="checkbox"
          tabIndex="1"
          data-name={this.props.name}
          aria-checked={this.isChecked().toString()}
        >
          <div>
            <div
              className={this.getElementName('checkbox', 'icon', {
                checked: true
              })}
            >
              <Icon size={this.props.size} name='checkbox-checked' />
            </div>
            <div
              className={this.getElementName('checkbox', 'icon', {
                notChecked: true
              })}
            >
              <Icon size={this.props.size} name='checkbox' />
            </div>
          </div>
          <div>{this.props.children}</div>
        </div>
      );
    }

    return (
      <div
        className={this.getBlockName('checkbox', this.getModifierObject())}
        data-name={this.props.name}
        aria-checked={this.isChecked().toString()}
        tabIndex="1"
        onClick={(e) => this.onChange(e)}
      >
        <div className={this.getElementName('checkbox', 'icon')}>
          {this.props.checked ? (
            <Icon
              size={this.props.size}
              name="checkbox"
            />
          ) : null}
        </div>
        <div>{this.props.children}</div>
      </div>
    )
  }
}
