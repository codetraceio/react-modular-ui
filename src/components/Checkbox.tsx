import * as React from "react";

import settings from "../services/settingService";

import { IModifiers, getBlockName, getElementName } from "../services/componentService";

export interface ICheckboxProps {
  size?: string | number;
  view?: string;
  color?: string;
  disabled?: boolean;
  checked?: boolean;
  name?: string;

  onChange?: (checked: boolean, event: React.MouseEvent<HTMLDivElement>) => void;
}

import Icon from "./Icon";

export default class Button extends React.PureComponent<ICheckboxProps, {}> {
  getModifierObject(): IModifiers {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      disabled: this.props.disabled
    };
  }

  isChecked(): boolean {
    return this.props.checked || false;
  }

  onChange = (e: React.MouseEvent<HTMLDivElement>) => {
    if (typeof this.props.onChange === "function") {
      this.props.onChange(!this.props.checked, e);
    }
  };

  render() {
    if (settings.isBackend()) {
      return (
        <div
          className={getBlockName("checkbox", this.getModifierObject())}
          role="checkbox"
          tabIndex={1}
          data-name={this.props.name}
          aria-checked={this.isChecked().toString()}
        >
          <div>
            <div
              className={getElementName("checkbox", "icon", {
                checked: true
              })}
            >
              <Icon size={this.props.size} name="checkbox-checked"/>
            </div>
            <div
              className={getElementName("checkbox", "icon", {
                notChecked: true
              })}
            >
              <Icon size={this.props.size} name="checkbox"/>
            </div>
          </div>
          <div>{this.props.children}</div>
        </div>
      );
    }

    return (
      <div
        className={getBlockName("checkbox", this.getModifierObject())}
        data-name={this.props.name}
        aria-checked={this.isChecked().toString()}
        tabIndex={1}
        onClick={this.onChange}
      >
        <div className={getElementName("checkbox", "icon")}>
          {this.props.checked ? (
            <Icon
              size={this.props.size}
              name="checkbox"
            />
          ) : null}
        </div>
        <div>{this.props.children}</div>
      </div>
    );
  }
}
