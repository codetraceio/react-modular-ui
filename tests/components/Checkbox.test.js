import * as React from "react";
import * as renderer from "react-test-renderer";

import Checkbox from "../../src/components/Checkbox";

describe("Checkbox", () => {
  it("should render Checkbox", () => {
    const tree = renderer.create(
      <Checkbox
        size="32"
        view="flat"
        color="primary"
        disabled
      >Something</Checkbox>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
