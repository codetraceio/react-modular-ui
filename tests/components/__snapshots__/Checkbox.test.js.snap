// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render Checkbox 1`] = `
<div
  aria-checked="false"
  className="c-checkbox -size-32 -view-flat -color-primary -disabled"
  data-name={undefined}
  onClick={[Function]}
  tabIndex="1"
>
  <div
    className="c-checkbox--icon"
  />
  <div>
    Something
  </div>
</div>
`;
